File: apps/demo/modules/expo-module-showcase/ios/ExpoModuleShowcaseConsumerModule.swift
import ExpoModulesCore

public final class ExpoModuleShowcaseConsumerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoModuleShowcaseConsumer")
    Function("readSharedCounter") { (counter: ShowcaseSharedCounter) in counter.value }
    Function("forwardSharedCounter") { (counter: ShowcaseSharedCounter) in counter }
    Function("readSharedTextRef") { (ref: ShowcaseTextRef) in ref.ref }
    Function("forwardSharedTextRef") { (ref: ShowcaseTextRef) in ref }
  }
}

File: apps/demo/modules/expo-module-showcase/android/src/main/java/expo/modules/showcase/ExpoModuleShowcaseConsumerModule.kt
package expo.modules.showcase

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoModuleShowcaseConsumerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoModuleShowcaseConsumer")
    Function("readSharedCounter") { counter: ShowcaseSharedCounter -> counter.value }
    Function("forwardSharedCounter") { counter: ShowcaseSharedCounter -> counter }
    Function("readSharedTextRef") { ref: ShowcaseTextRef -> ref.ref }
    Function("forwardSharedTextRef") { ref: ShowcaseTextRef -> ref }
  }
}
